### PR TITLE
Automatedtesting

### PR DESCRIPTION
**Pull request by KyleBehiels**
End to end testing is complete. Every push does the following:

- Checks if data has been cached
    -If so, run an abbreviated install to save time and not stress the server
    -Otherwise, run full base install
-Runs the test_metannotate.sh script
-Verifies that the test_output files match the reference files in testing/test_constants

### CODE REVIEW

File: testing/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from modules import *
```
Comment (LeeBergstrand) [inline testing/__init__.py]: @KyleBehiels You could use relative import syntax to import the modules as well. For example for in ```test_hash.py``` you could use.

```python
from modules.hash import hash
```
This is more explicit as the user upgrading the tests can see here the has function is located (i.e modules --> hash.py). 

https://stackoverflow.com/questions/72852/how-to-do-relative-imports-in-python

Comment (LeeBergstrand) [reply testing/__init__.py]: In summery remove this.

File: testing/base_installation_for_test.sh [hunk 2]
```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
```
Comment (LeeBergstrand) [inline testing/base_installation_for_test.sh]: @KyleBehiels If we can we can we should just use the standard install script. Therefore we don't have to go back and change both install scripts one after the other. For example the pip installs below need to be changed in both files. **If their is specific install commands we need to skip for testing we can use a shell argument and some if statements to skip certain steps in the install process.** Its could to test the base install in your CI system as it makes sure that the install will work every time.

Comment (LeeBergstrand) [reply testing/base_installation_for_test.sh]: Use the actually base install script in the root directory.

**Review by LeeBergstrand — CHANGES_REQUESTED** [review]

**Comment by LeeBergstrand** [comment]
@KyleBehiels I think a much better approach would be to have the circle-ci config install the software. Then run the tests rather than have the test install the software.

**Comment by LeeBergstrand** [comment]
@KyleBehiels https://circleci.com/docs/1.0/installing-custom-software/

**Comment by LeeBergstrand** [comment]
@KyleBehiels My company uses travis-ci for testing software. They have a bit better tutorial to explains this than circle-ci. https://docs.travis-ci.com/user/customizing-the-build

Same idea but different key words.

**Comment by KyleBehiels** [comment]
Thanks, I'll work on it!

**Pull request closed without merging**